src/app/model/migrations/migrations/to0002/MigrationToVersion0002.ts

```diff
@@ -0,0 +1,212 @@
+      const { appletId, entityId, eventId } = progressFlowFrom;
+
+      const logAppletName = '',
+        logFlowName = '';
+      const logFlowStateFrom = '';
+      const logCurrentActivityDto = '';
+      const logActivityFlowDto = '';
+      const logProgressFlowFrom = JSON.stringify(progressFlowFrom, null, 2);
+
+      try {
+        const key = this.getFlowRecordKey(appletId, entityId, eventId);
+
+        const flowStateFrom = this.getFlowState(key)!;
+
+        if (!flowStateFrom) {
+          Logger.warn(
+            `[MigrationToVersion0002.getUpdatedProgressForFlows]: Migration cannot be executed as flowState doesn't exist appletName=${logAppletName}, appletId=${appletId}, entityId=${entityId}, eventId=${eventId}`,
+          );
+          continue;
+        }
+        const flowName = flowStateFrom.flowName;
+
+        const progressFlowTo = this.getUpdatedFlowProgress(
+          progressFlowFrom,
+          flowName,
+        );
+
+        progressFlowsTo.push(progressFlowTo);
+      } catch (error) {
+        Logger.warn(
+          `[MigrationToVersion0002.getUpdatedProgressForFlows]: Error occurred, appletName=${logAppletName}, flowName=${logFlowName}, progressFlowFrom=${logProgressFlowFrom}, flowStateFrom=${logFlowStateFrom}, currentActivityDto=${logCurrentActivityDto}, activityFlowDto=${logActivityFlowDto}  \nerror: \n${error}`,
+        );
+      }
+    }
+
+    return progressFlowsTo;
+  }
+
+  private getUpdatedProgressForActivities(
+    reduxState: RootStateFrom,
+  ): NotCompletedEntitiesTo[] {
+    const progressActivitiesFrom = selectNotCompletedActivities(reduxState);
+    const progressActivitiesTo = [];
+
+    for (const progressActivityFrom of progressActivitiesFrom) {
+      const { appletId, entityId } = progressActivityFrom;
+      let logAppletName = '',
+        logActivityName = '';
+      const logActivityStateFrom = '';
+      const logCurrentActivityDto = '';
+      const logActivityActivityDto = '';
+      const logProgressActivityFrom = JSON.stringify(
+        progressActivityFrom,
+        null,
+        2,
+      );
+
+      try {
+        const appletDto = this.queryDataUtils.getAppletDto(appletId);
+
+        if (!appletDto) {
+          Logger.warn(
+            "[MigrationToVersion0002.getUpdatedProgressForActivities]: Migration cannot be executed as applet doesn't exist: " +
+              appletId,
+          );
+          continue;
+        }
+        logAppletName = appletDto.displayName;
+
+        const activityDto = appletDto.activities.find(f => f.id === entityId);
+
+        if (!activityDto) {
+          Logger.warn(
+            "[MigrationToVersion0002.getUpdatedProgressForActivities]: activityFlow doesn't exist: " +
+              entityId,
+          );
+          continue;
+        }
+
+        logActivityName = activityDto.name;
+
+        const progressActivityTo = this.getUpdatedActivityProgress(
+          progressActivityFrom,
+          activityDto.name,
```

Comment:
You should extract activityName from FlowState.pipeline (from flowRecord), almost the same as you do it in the getUpdatedProgressForFlows.